python/ql/test/experimental/dataflow/global-flow/test.py

```diff
@@ -0,0 +1,118 @@
+### Tests of global flow
+
+# Simple assignment
+
+g = [5] # $writes=g
+
+# Multiple assignment
+
+g1, g2 = [6], [7] # $writes=g1 $writes=g2
+
+# Assignment that's only referenced in this scope. This one will not give rise to a `ModuleVariableNode`.
+
+unreferenced_g = [8]
+print(unreferenced_g)
+
+# Testing modifications of globals
+
+# Modification by reassignment
+
+g_mod = []
+# Immediate reassignment -- no modification possible.
+g_mod = [10] # $f+:writes=g_mod
+print(g_mod) # Could potentially be redefined here.
```

Comment:
I didn't quite understand what was going on here, but I think I do now.

We're not considering `g_mod = []` a `ModuleVariableNode`, since we know it doesn't escape.

We do consider `g_mod = [10]` a `ModuleVariableNode`, since our analysis (wrongly) thinks it might escape to the `print()` call. This is a FP since we assume `print` won't change the value of `g_mod`.

```suggestion
g_mod = []
# The following assignment should not be a `ModuleVariableNode`, but currently our analysis thinks `g_mod` might be used in the `print` call
g_mod = [10] # $f+:writes=g_mod
print("foo")
```

_P.S. I checked using `"foo"` as the argument, and that does work :+1: and I think it's more clear what is going on when not using the actual variable as an argument_